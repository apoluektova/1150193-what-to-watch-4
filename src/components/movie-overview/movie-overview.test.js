import React from "react";
import renderer from "react-test-renderer";
import MovieOverview from "./movie-overview.jsx";

const movie = {
  previewImage: `img/aviator.jpg`,
  previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  title: `Aviator`,
  backgroundImage: `https://loremflickr.com/cache/resized/65535_49800111821_ae7805f489_h_1280_543_nofilter.jpg`,
  poster: `https://loremflickr.com/cache/resized/65535_50063687282_2595a7661e_z_273_410_nofilter.jpg`,
  genre: `Drama`,
  releaseDate: 2004,
  description: `Based on the 1993 non-fiction book Howard Hughes: The Secret Life by Charles Higham, the film depicts the life of Howard Hughes, an aviation pioneer and director of Hell's Angels. The film portrays his life from 1927–1947 during which time Hughes became a successful film producer and an aviation magnate while simultaneously growing more unstable due to severe obsessive–compulsive disorder (OCD).`,
  rating: {
    score: 8.7,
    level: `Very good`,
    count: 241,
  },
  director: `Martin Scorsese`,
  actors: `Leonardo DiCaprio, Cate Blanchett, and Kate Beckinsale`,
  runtime: `1h 39m`,
};

it(`MovieOverview should render correctly`, () => {
  const tree = renderer
     .create(<MovieOverview
       movie={movie}
     />)
     .toJSON();

  expect(tree).toMatchSnapshot();
});
