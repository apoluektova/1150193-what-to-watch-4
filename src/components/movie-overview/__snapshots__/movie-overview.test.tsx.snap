// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieOverview should render correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        234
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    The story follows a Scottish general's rise to power after receiving a prophecy from a trio of witches that one day he will become King of Scotland. Like the play it was adapted from, the film dramatises the damaging physical and psychological effects of political ambition on those who seek power for its own sake.
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Justin Kurzel
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Michael Fassbender, Marion Cotillard, Paddy Considine, Sean Harris
         and other
      </strong>
    </p>
  </div>,
]
`;
